### ISSUE TITLE
insight can't fetch history (or even balance) for addresses with large number of transactions

### ISSUE THREAD

**Issue by tphyahoo** [bitpay/insight#492]
Insight times out when trying to fetch ransaction history for address with large number of transactions, such as a satoshi dice address.

````
  http://live.insight.is/api/txs?address=1dice9wVtrKZTBbAZqz1XiTmboYyvpD3t 
  ( TIMES OUT )
````

FWIW, obelisk blockchain database from unsystem (accessed via sx below) can fetch the balance but also times out on history. 

root@ffob:/deploy# time sx balance 1dice9wVtrKZTBbAZqz1XiTmboYyvpD3t
Address: 1dice9wVtrKZTBbAZqz1XiTmboYyvpD3t
  Paid balance:    18241421
  Pending balance: 18241421
  Total received:  2837644402034

Blockchain.info blockchain database seems to be able to fetch all infos, paginated of course.

https://blockchain.info/address/1dice9wVtrKZTBbAZqz1XiTmboYyvpD3t

Possibly fixing this is as simple as adding pagination to insight / insight api. Possibly it also requires caching balance somewhere.

I know that counterparty uses insight for its block explorer. If it also uses insight for its transaction engine, I suppose this might actually be causing problems. I don't think that's the case, but I think it might be worth a sanity check.

**Comment by braydonf** [comment]
This should be improved, and addressed, with the v0.3.0 branch with the related insight-api v0.3.0 and bitcore-node.

**Comment by braydonf** [comment]
This has been improved in the latest master branch. However there are areas that could also use improvements: https://github.com/bitpay/bitcore-node/issues/354
